### PR TITLE
Added VIF STCYCL test.

### PR DESCRIPTION
**Pull request by jpd002**
Simple test for VIF's read/write cycles behaviors. There's tests for normal (or "full") unpacks and also for "truncated" unpacks, which are basically unpacks that don't complete in a single DMA transfer. I added the "truncated" cases because I wanted to see how the hardware reacts when it resumes a paused unpack command.

### CODE REVIEW

File: tests/dma/vif/stcycl.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+#include "vifregs.h"
+
+static u8 *const vu0_mem = (u8 *)0x11004000;
+
+void testCycleNormal(u32 cl, u32 wl) {
+	volatile u32 *vu0_32 = (volatile u32 *)vu0_mem;
+	
+	static const u32 outputSize = 8;
+	static const u32 displaySize = 64;
+	static const u32 vifPacketSize = (outputSize + 4);
```
Comment (unknownbrackets) [inline tests/dma/vif/stcycl.cpp:18]: Hmm, I should probably add a `vifcode.Qwords()` or something...

-[Unknown]

**Comment by unknownbrackets** [comment]
Is it worth testing upper bounds of the cl or wl values?

-[Unknown]

**Comment by jpd002** [comment]
Oh yeah, they might be. They are 8-bits values, so, it might get interesting if they are set to 255. I'll do the changes and update the branch.

**Comment by jpd002** [comment]
Alright, I've added tests for large values of CL and WL. Large CL values do exhibit an interesting behavior, probably because the write addresses are masked.

**Comment by unknownbrackets** [comment]
Actually, 000003fe seems to imply that it's signed, right?  Or something.  Definitely interesting behavior...

-[Unknown]

**Comment by jpd002** [comment]
0x03FE is actually WL (0x03) and CL (0xFE) packed together in a 16-bits value. I did some tests on my side and it's just the write addresses that are wrapping around. I'm thinking of changing this test to execute it on both VIF0 and VIF1 because they have different VU memory sizes (4KB for VU0 and 16KB for VU1).

**Comment by unknownbrackets** [comment]
Right, I meant that it almost seemed like CL was being treated as -2, but I see what you're saying, it's just wrapping around and giving that impression.  Sorry.

That's a good point about the mem size, just to make it clearer, and verify the behave the same.

-[Unknown]

**Pull request merged**